spot_hardware_interface/src/spot_hardware_interface.cpp

```diff
@@ -269,24 +247,33 @@ hardware_interface::return_type SpotHardware::read(const rclcpp::Time& /*time*/,
   }
   // Ensure that the states received from the Spot SDK will fit into the hw_states_ vector
   const auto states_size = hw_states_.size();
-  if (interfaces_per_joint_ * joint_pos.size() != states_size ||
-      interfaces_per_joint_ * joint_vel.size() != states_size ||
-      interfaces_per_joint_ * joint_load.size() != states_size) {
+  if (state_interfaces_per_joint_ * joint_pos.size() != states_size ||
+      state_interfaces_per_joint_ * joint_vel.size() != states_size ||
+      state_interfaces_per_joint_ * joint_load.size() != states_size) {
     RCLCPP_FATAL(
         rclcpp::get_logger("SpotHardware"),
         "The number of joints and interfaces does not match with the outputted joint states from the Spot SDK!");
     return hardware_interface::return_type::ERROR;
   }
   // Read values into joint states
   for (size_t i = 0; i < joint_pos.size(); ++i) {
-    hw_states_.at(i * interfaces_per_joint_) = joint_pos.at(i);
-    hw_states_.at(i * interfaces_per_joint_ + 1) = joint_vel.at(i);
-    hw_states_.at(i * interfaces_per_joint_ + 2) = joint_load.at(i);
+    hw_states_.at(i * state_interfaces_per_joint_) = joint_pos.at(i);
+    hw_states_.at(i * state_interfaces_per_joint_ + 1) = joint_vel.at(i);
+    hw_states_.at(i * state_interfaces_per_joint_ + 2) = joint_load.at(i);
   }
 
-  // Set initial command values to current state
+  // Fill in the initial command values
   if (!init_state_) {
-    hw_commands_ = hw_states_;
+    for (size_t i = 0; i < njoints_; i++) {
+      // copy over current position, velocity, and load from state to current command
+      hw_commands_[command_interfaces_per_joint_ * i] = hw_states_[state_interfaces_per_joint_ * i];
+      hw_commands_[command_interfaces_per_joint_ * i + 1] = hw_states_[state_interfaces_per_joint_ * i + 1];
+      hw_commands_[command_interfaces_per_joint_ * i + 2] = hw_states_[state_interfaces_per_joint_ * i + 2];
+      // Fill in k_q_p and k_qd_p gains from the initial_value field from the URDF
+      const auto& joint = info_.joints.at(i);
+      hw_commands_[command_interfaces_per_joint_ * i + 3] = std::stof(joint.command_interfaces.at(3).initial_value);
+      hw_commands_[command_interfaces_per_joint_ * i + 4] = std::stof(joint.command_interfaces.at(4).initial_value);
```

Comment:
nit: we could replace the hard-coded offsets with `pos_offset = 0, vel_offset = 1, load_offset = 2, ...` to make it easier to parse and don't assume or rely on remembering a known order in the code.